syntax = "proto3";
import "google/protobuf/empty.proto";
option go_package = ".;proto";

service Address{
  rpc CreateAddress(AddressRequest) returns (AddressResponse);//添加地址
  rpc GetAddress(AddressRequest) returns (AddressListResponse);//获取地址
  rpc UpdateAddress(AddressRequest) returns (google.protobuf.Empty);//更新地址
  rpc DeleteAddress(AddressRequest) returns (google.protobuf.Empty);//删除地址

}
message AddressRequest{
  int32 id = 1;
  int32 userId = 2;
  string province = 3;
  string city = 4;
  string district = 5;
  string address = 6;
  string signerName = 7;
  string signerPhone = 8;
}
message AddressResponse{
  int32 id = 1;
  int32 userId = 2;
  string province = 3;
  string city = 4;
  string district = 5;
  string address = 6;
  string signerName = 7;
  string signerPhone = 8;
}
message AddressListResponse {
  int32 total = 1;
  repeated AddressResponse data = 2;
}